File: java/adhan/src/main/java/com/batoulapps/adhan/Madhab.java
package com.batoulapps.adhan;

import com.batoulapps.adhan.internal.ShadowLength;

/**
 * Madhab for determining how Asr is calculated
 */
public enum Madhab {
  /**
   * Shafi Madhab
   */
  SHAFI,

  /**
   * Hanafi Madhab
   */
  HANAFI;

  ShadowLength getShadowLength() {
    switch (this) {
      case SHAFI: {
        return ShadowLength.SINGLE;
      }
      case HANAFI: {
        return ShadowLength.DOUBLE;
      }
      default: {
        throw new IllegalArgumentException("Invalid Madhab");
      }
    }
  }
}

File: Swift/Sources/Enums/Madhab.swift
//
//  Madhab.swift
//  Adhan
//
//  Copyright © 2018 Batoul Apps. All rights reserved.
//

import Foundation

/* Madhab for determining how Asr is calculated */
public enum Madhab {
    case shafi
    case hanafi

    var shadowLength: ShadowLength {
        switch(self) {
        case .shafi:
            return .single
        case .hanafi:
            return .double
        }
    }
}
